### PR TITLE
Test-enable XIO messenger and IB/RDMA support

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: http://tracker.ceph.com/issues/15472

Signed-off-by: Lars Marowsky-Bree <lmb@suse.com>
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
https://github.com/ceph/ceph/blob/83e10f7e2df0a71bd59e6ef2aa06b52b186fddaa/README.xio

File: ceph.spec.in [hunk 1]
```diff
@@ -150,6 +151,12 @@ BuildRequires:	libbz2-devel
 %if 0%{with tcmalloc}
 BuildRequires:	gperftools-devel
 %endif
+%if 0%{with xio}
+BuildRequires:  librdmacm-devel
+BuildRequires:  libibverbs-devel
+BuildRequires:  libnuma-devel
+BuildRequires:  libevent-devel
```
Comment (roidayan) [inline ceph.spec.in]: actually those are Accelio requirements. so the requirements should actually be in libxio.spec (accelio package name).
ceph requirement should be libxio-devel for build and libxio for install.

Comment (ktdreyer) [inline ceph.spec.in]: Yeah, if that's the case, it does sound like an libxio packaging bug...

Comment (mattbenjamin) [inline ceph.spec.in]: @ktdreyer  are we ready to merge this?  @liewegas, I guess this can't be Jewel?

**Comment by smithfarm** [comment]
@roidayan So, like this?

SUSE does not yet have an accelio package, but I will make one. @ktdreyer RH/CentOS does have it, though, yes?

**Comment by liewegas** [comment]
Prefer to wait until after jewel, unless this is 0 risk on the packaging side.

**Comment by roidayan** [comment]
@smithfarm yes thanks

**Comment by smithfarm** [comment]
@roidayan We are going to do some testing on this soon, but perhaps you already know whether libxio is required on clients (i.e. for the ceph-common package) as well as servers?

**Comment by roidayan** [comment]
@smithfarm yes. clients will want to use xio messenger as well.

**Comment by smithfarm** [comment]
OK, updated the patch so it adds the `libxio` runtime dependency to both `ceph-base` and `ceph-common`

**Comment by smithfarm** [comment]
We have this in a build that we are currently testing on real hardware.

**Comment by liewegas** [comment]
Ok to merge when you're satisfied, I think!

**Comment by smithfarm** [comment]
Rebased

**Pull request merged**